libs/pages/application/src/lib/feature/page-settings-general-feature/page-settings-general-feature.tsx

```diff
@@ -116,148 +103,130 @@ export const handleJobSubmit = (data: FieldValues, application: ApplicationEntit
 
   const { data: organization } = useOrganization({ organizationId })
-  const loadingStatus = useSelector((state: RootState) => getApplicationsState(state).loadingStatus)
+  const { data: service } = useService({ environmentId, serviceId: applicationId })
+  const { mutate: editService, isLoading: isLoadingEditService } = useEditService({ environmentId })
+
+  const helmRepository = match(service)
+    .with({ serviceType: 'HELM', source: P.when(isHelmRepositorySource) }, ({ source }) => source.repository)
+    .otherwise(() => undefined)
+  const helmGit = match(service)
+    .with({ serviceType: 'HELM', source: P.when(isHelmGitSource) }, ({ source }) => source.git?.git_repository)
+    .otherwise(() => undefined)
 
   const methods = useForm({
     mode: 'onChange',
+    defaultValues: {
+      name: service?.name,
+      description: service?.description,
+      auto_deploy: (service as Application)?.auto_deploy,
+      build_mode: service?.serviceType === 'JOB' ? BuildModeEnum.DOCKER : (service as Application)?.build_mode,
+      buildpack_language: (service as Application)?.buildpack_language ?? BuildPackLanguageEnum.PYTHON,
+      dockerfile_path: (service as Application)?.dockerfile_path ?? 'Dockerfile',
+      registry: (service as Container)?.registry?.id,
+      image_name: (service as Container)?.image_name,
+      image_tag: (service as Container)?.tag,
+      image_entry_point: (service as Application)?.entrypoint,
+      cmd_arguments:
+        (service as Application)?.arguments && (service as Application)?.arguments?.length
+          ? JSON.stringify((service as Application).arguments)
+          : '',
+      source_provider: isHelmRepositorySource((service as Helm)?.source) ? 'HELM_REPOSITORY' : 'GIT',
+      repository: helmRepository?.repository?.id ?? helmGit?.url,
+      chart_name: helmRepository?.chart_name,
+      chart_version: helmRepository?.chart_version,
+    },
   })
 
-  const watchBuildMode = methods.watch('build_mode')
-
-  const toasterCallback = () => {
-    if (application) {
-      dispatch(
-        postApplicationActionsRedeploy({
-          applicationId,
-          environmentId,
-          serviceType: getServiceType(application),
-          callback: () =>
-            navigate(
-              ENVIRONMENT_LOGS_URL(organizationId, projectId, environmentId) + DEPLOYMENT_LOGS_URL(applicationId)
-            ),
-          queryClient,
-        })
-      )
-    }
-  }
-
   const onSubmit = methods.handleSubmit((data) => {
-    if (data && application) {
-      let cloneApplication: Omit<ApplicationEntity, 'registry'> & { registry?: { id?: string } }
-      if (isApplication(application)) {
-        cloneApplication = handleGitApplicationSubmit(data, application)
-      } else if (isJob(application)) {
-        cloneApplication = handleJobSubmit(data, application)
-      } else {
+    if (!service) return
+
+    return match(service)
```

Comment:
Build the `payload` with match then call `editService` only once

Reply:
I created a payload with `match` I had just one specific case with `Container` because I need to have a `ToastError`, so we have two time `editService`